components/ipfs/ipfs_utils.cc

```diff
@@ -425,4 +425,45 @@ std::string GetRegistryDomainFromIPNS(const GURL& url) {
       cid, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
 }
 
+// gateway.io/ipfs/bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy ->
+// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
+// bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy.ipfs.gateway.io ->
+// ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfy
+absl::optional<GURL> TranslateToCurrentGatewayUrl(const GURL& url) {
+  if (!url.is_valid() || !url.SchemeIsHTTPOrHTTPS()) {
+    return absl::nullopt;
+  }
+
+  std::vector<std::string> host_parts = base::SplitStringUsingSubstr(
+      url.host(), ".", base::KEEP_WHITESPACE, base::SPLIT_WANT_ALL);
+
+  if (host_parts.size() > 2 && IsValidCID(host_parts.at(0)) &&
+      host_parts.at(1) == "ipfs") {
+    GURL final_url = GURL("ipfs://" + host_parts.at(0) + url.path());
+    GURL::Replacements replacements;
+    replacements.SetQueryStr(url.query_piece());
```

Comment:
Shouldn't we do `replacements.SetRefStr(url.ref_piece());` too? Need to add test for this too if so.
Same question to the if block below.

Reply:
Fixed